1-Parallelize-your-python-code_Futures_and_Delayed.ipynb

```diff
@@ -0,0 +1,1062 @@
+   "metadata": {},
+   "source": [
+    "### Delayed \n",
+    "\n",
+    "Similarly to `futures`, `delayed` can be used to support arbitrary task scheduling, but in the case of `delayed` this happens **lazily**. This is the important difference between futures and delayed, `delayed` constructs a graph while `futures` are eager.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5cd4ab43-87d0-4e0f-984b-569c2716cad9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dask.delayed\n",
+    "def inc(x):\n",
+    "    sleep(1)\n",
+    "    return x + 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "91afd8d6-3f7a-4857-bb51-ba86ae34a341",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "x = inc(1)\n",
+    "x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "312dc42e-3795-402b-ab38-0584f6d45891",
+   "metadata": {},
+   "source": [
+    "This run immediately, since nothing has really happened yet.\n",
+    "\n",
+    "To get the result, call `compute`. Notice that this runs faster than the original code."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9bdfc2d6-212f-48f6-a224-f3a6b6f3d0d4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ec8b689c-a070-4faf-9bc9-8a0f702a403d",
+   "metadata": {},
+   "source": [
+    "The same example we did with `client.map()`, using `delayed` would be"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3f104ffe-c778-4ee5-91a6-14fb0bbded50",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "results = []\n",
+    "for x in range(8):\n",
+    "    y = inc(x)\n",
+    "    results.append(y)\n",
+    "    \n",
+    "total = sum(results)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6027cba4-39b2-46d2-b3ef-76d012f4d544",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "total.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb0ffe5d-6453-43f8-b92a-4b81edde93e6",
+   "metadata": {},
+   "source": [
+    "**NOTE:** For an introductory lesson on delayed revisit:\n",
+    "- https://tutorial.dask.org/03_dask.delayed.html\n",
+    "\n",
+    "**Useful links: delayed**\n",
+    "\n",
+    "* [Delayed documentation](https://docs.dask.org/en/latest/delayed.html)\n",
+    "* [Delayed screencast](https://www.youtube.com/watch?v=SHqFmynRxVU)\n",
+    "* [Delayed API](https://docs.dask.org/en/latest/delayed-api.html)\n",
+    "* [Delayed examples](https://examples.dask.org/delayed.html)\n",
+    "* [Delayed best practices](https://docs.dask.org/en/latest/delayed-best-practices.html)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "16467916-008c-4a21-bb1e-d374c530ee57",
+   "metadata": {},
+   "source": [
+    "## Grown-up example: Scrapping and cleaning SO data\n",
```

Comment:
I like that the stack overflow example is concrete/grown up 👍  

I don't understand the walk around part.  This feels not relevant to learning Dask.  Instead I recommend giving the function and demonstrating it, but not going into depth.  This doesn't seem like what we want them thinking about.